### PR TITLE
JP-3543: Allow alternate resampling weight type in outlier detection

### LINKED ISSUE #8289: Outlier detection overrides weight_type parameter
_Issue [JP-3543](https://jira.stsci.edu/browse/JP-3543) was created on JIRA by [Melanie Clarke](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mclarke):_

The outlier_detection step currently accepts a weight_type parameter, but overrides or fails to pass it along in several places. 

Testing with NIRSpec FS and MOS data, the default value of 'ivm' is always used, even when 'exptime' is passed as an override.  For bright sources, ivm weights are a poor match to the data and we'd like to be able to override the weight_type for better outlier detection.

### PR DESCRIPTION
**Pull request by melanieclarke**
Resolves [JP-3543](https://jira.stsci.edu/browse/JP-3543)

Closes #8289 

This PR passes the input weight_type parameter to all the resampling functions called within the outlier_detection step.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by tapastro** [comment]
Regtests: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1220/

**Comment by melanieclarke** [comment]
I think the regression test failures are unrelated.  The MIRI tests pass locally on this branch with current truth files.  The failed NIRCam test is also failing on main.

**Review by emolter — APPROVED** [review]
Looks good to me

**Review by hbushouse — APPROVED** [review]
Looks OK to me.

File: jwst/outlier_detection/outlier_detection_step.py [hunk 1]
```diff
@@ -102,7 +102,8 @@ def process(self, input_data):
 
             # Setup outlier detection parameters
             pars = {
-                'weight_type': self.weight_type,
+                'weight_type': self.weight_type,  # for calling the resample step
+                'wht_type': self.weight_type,  # for calling the resample class directly
```
Comment (hbushouse) [inline jwst/outlier_detection/outlier_detection_step.py:106]: @melanieclarke This list is strictly internal use only, right?  i.e. we're not going to be presenting users with a confusing set of names of both `weight_type` and `wht_type`

Comment (melanieclarke) [reply jwst/outlier_detection/outlier_detection_step.py:106]: Yes, this is only used internally.  The top-level parameter is "weight_type" only.

**Comment by hbushouse** [comment]
Started another regtest run at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1394

**Pull request merged**